common/lib/client/auth.js

```diff
@@ -534,6 +519,144 @@ var Auth = (function() {
 		}
 	};
 
+	Auth.prototype._saveBasicOptions = function(defaultTokenParams, authOptions) {
+		this.method = 'basic';
+		this.key = authOptions.key;
+		this.basicKey = toBase64(authOptions.key);
+		this.tokenParams = defaultTokenParams || {};
+		this.authOptions = authOptions || {};
+		delete this.authOptions.force;
+		if('clientId' in authOptions) {
+			this._userSetClientId(authOptions.clientId);
+		}
+	}
+
+	Auth.prototype._saveTokenOptions = function(tokenParams, authOptions) {
+		this.method = 'token';
+
+		this.tokenParams = tokenParams || this.tokenParams || {};
+
+		/* If an authOptions object is passed in that contains new auth info (ie
+		* isn't just {force: true} or something), it becomes the new default, with
+		* the exception of the force attribute (RSA10g), which is set anew on each
+		* call to authorise (defaulting to false) */
+		this.force = false;
+		if(authOptions) {
+			this.force = authOptions.force;
+
+			if(containsAuthInfo(authOptions)) {
+				this.authOptions = authOptions || {};
```

Comment:
I disagree unfortunately and believe we have merged in all other implementations.  I think a common use case would be to initialise the library, and then later add a `clientId` for an explicit authorise or change the `ttl`.  Assuming therefore the user did the following, I assume it would effectively nullify the `clientId` and the `authUrl`.

```javascript
client = new Ably.Realtime({ clientId: 'bob', authUrl: '/auth' });
client.authorise({ ttl: 60 }, {});
```

As we have defaults currently that should not be overwritten with each call to authorise, I personally prefer that if a user wants to remove an option they provide a `null` value which would work fine with a merge. Either way, we need to update the spec, but if we don't support merge, we're going to need to modify Python, PHP, Ruby, and iOS I believe.  I am not sure how Java approaches it.

@paddybyers thoughts?